hnn_core/pyramidal.py

```diff
@@ -53,13 +53,24 @@ class Pyr(_Cell):
 
     def __init__(self, pos, celltype, override_params=None, gid=None):
 
-        if celltype == 'L5_pyramidal':
+        if self.__name__ == 'L5_pyramidal':
```

Comment:
Why did you make this change? Seems `celltype` is what you want to test here?

Reply:
I think we had a discussion with @samikane that the `celltype` was kind of superfluous to pass in this function. But I agree, it's better not to change stuff that's not related to your pull request. Also, for this to work it should be:

```py
if self.__name__ == 'L5Pyr':
```